### PR TITLE
Add generalized regular grid (Reciprocal mode)

### PR DESCRIPTION
**Pull request by atztogo**
"Reciprocal" mode of the generalized regular grid has been added.
https://www.vasp.at/wiki/index.php/KPOINTS#Generalized_regular_meshes
There is another mode "Cartesian". I only implemented "Reciprocal" mode because I don't find a good reason to use "Cartesian" mode. In addition, the keyword "Cartesian" is used in "Explicit k-point list", and I am afraid of breaking the existing logic in parsevasp.

I ran pre-commit hook but could not solve error because it was raised at the line that I didn't write. So I committed with `--no-verify`.

### CODE REVIEW

**Review by espenfl — COMMENTED** [review]
@atztogo Looks good. Thanks for adding it. I just had some minor comments.

File: parsevasp/kpoints.py [hunk 1]
```diff
@@ -330,6 +330,8 @@ def modify(self, entry, value, point_number=None):
                 self._check_comment(comment=value)
             if entry == 'divisions':
                 self._check_divisions(divisions=value)
+            if entry == 'gen_lat_vecs':
```
[the remark this answers is not in the document]
Comment (espenfl) [reply parsevasp/kpoints.py:333]: What about just `vectors` or even better possibly `generating_vectors`. Then we do not touch `lattice` at all. More general, so not more descriptive, but maybe less confusing. Yes, I noticed you chose the same wording as in the VASP docs. Other suggestions welcome.

[the remark this answers is not in the document]
Comment (atztogo) [reply parsevasp/kpoints.py:333]: @espenfl, thanks. Let's take `generating_vectors`. I will work for it tomorrow.

**Comment by atztogo** [comment]
@espenfl, I renamed `gen_lat_vecs` to `generating_vectors`. Could you have a look at pre-commit failure, https://github.com/aiida-vasp/parsevasp/runs/7142685054?check_suite_focus=true ?

**Comment by espenfl** [comment]
> @espenfl, I renamed `gen_lat_vecs` to `generating_vectors`. Could you have a look at pre-commit failure, https://github.com/aiida-vasp/parsevasp/runs/7142685054?check_suite_focus=true ?

@atztogo That is great. Thanks. We do not have a `.pylintrc`. Some linting need to be adapter to our needs. I see no real need to divide `kpoints.py` at this point, so please add a `.pylintrc` file with `disable=too-many-lines`. Then we would not get this for any file.

**Comment by espenfl** [comment]
Thanks a lot @atztogo.

**Pull request merged**